### PR TITLE
Feat(eos_cli_config_gen): Support ND parameters inside router_l2_vpn the same as ARP

### LINKED ISSUE #2537: Support ND parameters inside router_l2_vpn the same as ARP
### Enhancement summary

Should allow to configure these:
```
router l2-vpn
   nd learning bridged
   nd proxy prefix-list ND_PFX_LIST
```

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

IPv6 neighbor discovery in L2 VPN.

### Describe the solution you would like

Support the parameters for ND the same way ARP is already supported.

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by wnagele**
## Change Summary

Support the parameters for ND the same way ARP is already supported.

## Related Issue(s)

Fixes #2537

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
See PR.

## How to test
Added those config options and validated that they work the same way ARP does already.

### User Checklist

None.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-l2-vpn.yml [hunk 1]
```diff
@@ -7,3 +7,6 @@ router_l2_vpn:
   arp_proxy:
     prefix_list: "pl-router-l2-vpn"
   arp_learning_bridged: true
+  nd_proxy:
+    prefix_list: "pl-router-l2-vpn"
+  nd_learning_bridged: true
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-l2-vpn.yml]: Please add a newline at the end of the file.

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-l2-vpn.yml]: Other than that everything looks good to me.

**Comment by wnagele** [comment]
Ready for review and CI run for validation.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_l2_vpn.schema.yml [hunk 2]
```diff
@@ -16,6 +16,14 @@ keys:
...
             type: str
             description: Prefix-list Name
+      nd_learning_bridged:
+        type: bool
+      nd_proxy:
+        type: dict
+        keys:
+          prefix_list:
+            type: str
+            description: Prefix-list Name
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_l2_vpn.schema.yml]:
```suggestion
            description: Prefix-list name. VXLAN ND Proxying is disabled for IPv6 addresses defined in the prefix-list.
```

**Review by ClausHolbechArista — APPROVED** [review]
Only a minor schema description update requested. Other than that, LGTM. Thank you for another great contribution.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_l2_vpn.schema.yml (hunk 2 above)
Comment (wnagele) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_l2_vpn.schema.yml]: Done and also updated the ARP one.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-l2-vpn.j2 [hunk 3]
```diff
@@ -8,6 +8,12 @@ router l2-vpn
...
 {%     endif %}
+{%     if router_l2_vpn.nd_learning_bridged is arista.avd.defined(true) %}
+   nd learning bridged
+{%     endif %}
+{%     if router_l2_vpn.nd_proxy.prefix_list is arista.avd.defined %}
+   nd proxy prefix-list {{ router_l2_vpn.nd_proxy.prefix_list }}
+{%     endif %}
 {%     if router_l2_vpn.arp_selective_install is arista.avd.defined(true) %}
    arp selective-install
 {%     endif %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-l2-vpn.j2]:
```suggestion
{%     if router_l2_vpn.arp_selective_install is arista.avd.defined(true) %}
   arp selective-install
{%     endif %}
{%     if router_l2_vpn.nd_learning_bridged is arista.avd.defined(true) %}
   nd learning bridged
{%     endif %}
{%     if router_l2_vpn.nd_proxy.prefix_list is arista.avd.defined %}
   nd proxy prefix-list {{ router_l2_vpn.nd_proxy.prefix_list }}
{%     endif %}
```

Just a minor adjustment as per the eos cli
image

Comment (tgodaA) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-l2-vpn.j2]: `nd learning` and `nd proxy` comes after `arp selective-install`

Comment (wnagele) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-l2-vpn.j2]: This is just re-ordering, right?

Comment (tgodaA) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-l2-vpn.j2]: Yes. Nothing changed. After changing the template, re-generate the artifacts.

**Comment by wnagele** [comment]
I think this should be good for approval and merge now.

**Review by tgodaA — APPROVED** [review]
LGTM

**Pull request merged**